src/test/components/elements/inputs/RegistrationDobInput.test.tsx

```diff
@@ -0,0 +1,168 @@
+import { fireEvent, screen } from "@testing-library/react";
+import { renderTestEnvironment } from "../../../utils";
+import { RegistrationDobInput } from "../../../../app/components/elements/inputs/RegistrationDobInput";
+import { mockUserToUpdate } from "../../../../mocks/data";
+
+const mockSetUserToUpdate = jest.fn();
+const mockSetDobOver13CheckboxChecked = jest.fn();
+
+describe("RegistrationDobInput", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const setupTest = (
+    path: "/register/student" | "/register/teacher",
+    props = {}
+  ) => {
+    Object.defineProperty(window, "location", {
+      value: {
+        pathname: path,
+      },
+      writable: true,
+    });
+
+    renderTestEnvironment({
+      role: "ANONYMOUS",
+      PageComponent: RegistrationDobInput,
+      componentProps: {
+        userToUpdate: mockUserToUpdate,
+        setUserToUpdate: mockSetUserToUpdate,
+        submissionAttempted: false,
+        dobOver13CheckboxChecked: false,
+        setDobOver13CheckboxChecked: mockSetDobOver13CheckboxChecked,
+        ...props,
+      },
+      initialRouteEntries: ["/"],
+    });
+  };
+
+  it("renders correctly for student registration", () => {
+    setupTest("/register/student");
+    const dateOfBirthInputs = ["day", "month", "year"].map((unit) =>
```

Comment:
You could move this to a function (private to this file), since you've reused it a few times. The same goes for the `over13Checkbox`.